### PR TITLE
Add Basic Scalar Tests & Fix Scalar mistakes

### PR DESCRIPTION
**Pull request by HurricanKai**

### CODE REVIEW

**Comment by HurricanKai** [comment]
@Perksey please review :)
Note that this includes `net5.0` -> `netcoreapp5.0` because I really wanted a functional build, imo while official builds still use `netcoreapp5.0` this should too. (but I can change that back if you want me to)

**Comment by HurricanKai** [comment]
also note that CI is still failing because it's on .Net Core 3.1 and for some reason defaults to .net 5

File: src/Maths/Silk.NET.Maths/Silk.NET.Maths.csproj [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <Project Sdk="Microsoft.NET.Sdk">
     <PropertyGroup>
-        <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.0;net5.0</TargetFrameworks>
-        <DefineConstants Condition="'$(TargetFramework)' == 'netcoreapp3.0' Or '$(TargetFramework)' == 'net5.0'">
+        <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.0;netcoreapp5.0</TargetFrameworks>
```
Comment (Perksey) [inline src/Maths/Silk.NET.Maths/Silk.NET.Maths.csproj:3]: I'm against merging these TFM-related csproj changes. Ensure you have the latest .NET 5 preview and that you can build using `dotnet build`. I'd imagine IDEs aren't ready for prime time but the TFM should work regardless.

File: src/Maths/Silk.Net.Maths.Tests/Silk.Net.Maths.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.0;netcoreapp5.0</TargetFrameworks>
```
Comment (Perksey) [inline src/Maths/Silk.Net.Maths.Tests/Silk.Net.Maths.Tests.csproj:4]: `net5.0` here too. If you are experiencing issues with the latest preview (preview 5, use `dotnet --version` to check your current version) let me know.

**Pull request merged**